dl/src/main/scala/com/intel/analytics/bigdl/nn/Reverse.scala

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016 The BigDL Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.intel.analytics.bigdl.nn
+
+import com.intel.analytics.bigdl.nn.abstractnn.{AbstractModule, Activity}
+import com.intel.analytics.bigdl.tensor.Tensor
+import com.intel.analytics.bigdl.tensor.TensorNumericMath.TensorNumeric
+import com.intel.analytics.bigdl.utils.{T, Table}
+
+import scala.reflect.ClassTag
+
+/**
+ * Reverse the input w.r.t given dimension.
+ * The input can be a Tensor or Table.
+ * @param dim
+ * @param ev
+ * @tparam T Numeric type. Only support float/double now
+ */
+class Reverse[T: ClassTag](dim: Int = 1)
+  (implicit ev: TensorNumeric[T])
+  extends AbstractModule[Activity, Activity, T] {
```

Comment:
Reverse for Tensor, and ReverseTable for Table; this is what the naming convention looks like.